nemo/collections/llm/tools/autotuner/README.md

```diff
@@ -0,0 +1,421 @@
+- **Production Deployment**: Runs directly on customer GPU infrastructure, ready for enterprise use.
+
+
+
+# NeMo Autotuner Launcher
+
+A tool for orchestrating NeMo autotuner workflows on remote clusters using Lepton. This launcher handles packaging local code and data, then executes autotuner steps remotely.
+
+## Quick Start
+
+### Prerequisites
+- Installation
+
+   ```bash
+   # Ensure you have NeMo, nemo_run and rich installed
+   pip install nemo nemo_run rich
+   ```
+- NeMo workspace mounted at `/nemo-workspace`
+- Access to GPU resources on Lepton
+
+## Basic Usage
+
+### 1. Generate Configurations (`generate`)
+
+Creates optimized training configurations for your model and infrastructure.
+**Example**
+
+```bash
+python launcher.py generate \
+  --model llama2_7b \                                                       # Model name to generate configs for
+  --nodes 4 \                                                               # Number of compute nodes
+  --gpus-per-node 8 \                                                       # GPUs per node
+  --config-dir nemo-workspace/autotuner-data/autotuner-configs \            # Directory to save generated configs
+  --mount-path /workspace \                                                 # Remote mount path for workspace
+  --mount-from node-nfs:shared \                                            # Mount source for shared storage
+  --node-group gpu-cluster \                                                # Node group for resource allocation
+  --logs-subdir logs \                                                      # Subdirectory for training logs
+  --resource-shape gpu.8xh200 \                                             # GPU resource shape for training
+  --tensor-parallel-sizes 1,2,4,8 \                                         # Tensor parallel sizes (comma-separated)
+  --pipeline-parallel-sizes 1,2,4 \                                         # Pipeline parallel sizes (comma-separated)
+  --context-parallel-sizes 1,2 \                                            # Context parallel sizes (comma-separated)
+  --expert-parallel-sizes 1 \                                               # Expert parallel sizes (comma-separated)
+  --virtual-pipeline-parallel-sizes 1,2 \                                   # Virtual pipeline parallel sizes (comma-separated)
+  --global-batch-sizes 256,512,1024 \                                       # Global batch sizes (comma-separated)
+  --micro-batch-sizes 1,2,4,8 \                                             # Micro batch sizes (comma-separated)
+  --max-steps-per-run 10 \                                                  # Maximum steps per training run
+  --seq-length 8192 \                                                       # Sequence length for training
+  --num-tokens-in-b 15000 \                                                 # Number of tokens in billions
```

Comment:
I'd add that this is used just for calculating how long the actual model will be trained for.